velox/core/PlanNode.cpp

```diff
@@ -643,8 +643,18 @@ UnnestNode::UnnestNode(
   int unnestIndex = 0;
   for (const auto& variable : unnestVariables_) {
     if (variable->type()->isArray()) {
-      names.emplace_back(unnestNames[unnestIndex++]);
-      types.emplace_back(variable->type()->asArray().elementType());
+      bool isRowType = variable->type()->childAt(0)->isRow();
```

Comment:
@pramodsatya : Are arrays of arrays, arrays of maps etc allowed for Unnest ?

Reply:
@aditi-pandit, yes Presto permits this for Unnest and the current code is handling both these cases:
```
presto:tpch> select * from unnest(ARRAY[MAP(ARRAY[2, 4], ARRAY['3', '6']), map(array[2,3],array['ved', 'ghy'])], ARRAY[map(array[1,2,3],array[1.0,2.3,4.56])]) AS r(r1, r2);
       r1       |            r2
----------------+--------------------------
 {2=3, 4=6}     | {1=1.00, 2=2.30, 3=4.56}
 {2=ved, 3=ghy} | NULL

presto:tpch> select * from unnest(ARRAY[ARRAY[2, 4], ARRAY[3], ARRAY[2,3,4,5], ARRAY[4]], ARRAY[ARRAY[array[1,2],array[2],array[3,5,6]], ARRAY[array[4,5]]]) AS r(r1, r2);
      r1      |            r2
--------------+--------------------------
 [2, 4]       | [[1, 2], [2], [3, 5, 6]]
 [3]          | [[4, 5]]
 [2, 3, 4, 5] | NULL
 [4]          | NULL

presto:tpch> select * from unnest(ARRAY[ARRAY[2, 4], ARRAY[3], ARRAY[2,3,4,5], ARRAY[4]], ARRAY[ARRAY[1,2,3], ARRAY[4,5]]) AS r(r1, r2);
      r1      |    r2
--------------+-----------
 [2, 4]       | [1, 2, 3]
 [3]          | [4, 5]
 [2, 3, 4, 5] | NULL
 [4]          | NULL
```